src/indexer.c

```diff
@@ -113,6 +113,32 @@ void *ofi_idx_remove(struct indexer *idx, int index)
 	return item;
 }
 
+void *ofi_idx_remove_ordered(struct indexer *idx, int index)
+{
+	struct ofi_idx_entry *entry;
+	void *item;
+	int indx;
```

Comment:
nit: Please rename to cur_index or something similar to clearly indicate that it's not just a misspelled index.